src/Microsoft.OData.Edm/Csdl/CsdlJsonWriter.cs

```diff
@@ -54,6 +55,23 @@ protected override void WriteCsdl()
             WriteCsdlEnd();
         }
 
+        /// <summary>
+        /// Write the JSON CSDL asynchronously.
+        /// </summary>
+        /// <returns></returns>
+        protected override async Task WriteCsdlAsync()
+        {
+            await WriteCsdlStartAsync();

```

Comment:
We normally call [`ConfigureAwait(false)`](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.configureawait?view=net-8.0) to all the awaited async calls in our library. (I thought we had an analyzer to check this). You should do this across the async calls.
```suggestion
            await WriteCsdlStartAsync().ConfigureAwait(false);

```
Setting this to false, means the continuation of the awaited call does not have to wait to run on the same thread or context it was create from, it can continue on the next available thread. This can help improve perf and avoid deadlocks and is recommended practice inside library code.

To learn more, check this blog post: https://devblogs.microsoft.com/dotnet/configureawait-faq/

Reply:
It will save on performance, and it is usually best practice. I tried to research online if there is a default way of making ConfigureAwait false by default but that is not currently available. I will definitely add it to all of the awaits.

Reply:
@habbes Thanks for bringing this up. I have added `ConfigureAwait(false)` to all the awaited async calls.